app/src/main/java/com/android/voyageur/ui/utils/ImageCropperUtil.kt

```diff
@@ -0,0 +1,127 @@
+package com.android.voyageur.ui.utils
+
+import android.graphics.Bitmap
+import android.net.Uri
+import android.widget.Toast
+import androidx.activity.compose.rememberLauncherForActivityResult
+import androidx.activity.result.contract.ActivityResultContracts
+import androidx.compose.runtime.Composable
+import androidx.compose.runtime.remember
+import androidx.compose.ui.platform.LocalContext
+import com.canhub.cropper.CropImageContract
+import com.canhub.cropper.CropImageContractOptions
+import com.canhub.cropper.CropImageOptions
+import com.canhub.cropper.CropImageView
+
+/**
+ * Represents the result of an image cropping operation.
+ *
+ * @property imageUri The URI of the cropped image as a string, or null if cropping failed
+ * @property error Error message if the cropping operation failed, or null if successful
+ */
+data class ImageCropperResult(val imageUri: String?, val error: String?)
+
+/**
+ * A composable function that provides image cropping functionality with gallery integration. This
+ * utility allows users to either crop an existing image or select one from the gallery and then
+ * crop it.
+ *
+ * Features:
+ * - Customizable aspect ratio
+ * - Gallery integration
+ * - Error handling with user feedback
+ * - JPEG compression with quality control
+ * - Cropping guidelines
+ *
+ * Usage example:
+ * ```
+ * val imageCropper = rememberImageCropper(
+ *     aspectRatioX = 1,
+ *     aspectRatioY = 1
+ * ) { result ->
+ *     result.imageUri?.let { uri ->
+ *         // Handle the cropped image URI
+ *     }
+ *     result.error?.let { error ->
+ *         // Handle any errors
+ *     }
+ * }
+ *
+ * // Launch with existing URI
+ * imageCropper(existingUri)
+ *
+ * // Or launch gallery picker
+ * imageCropper(null)
+ * ```
+ *
+ * @param aspectRatioX The horizontal aspect ratio for cropping (default: 16)
+ * @param aspectRatioY The vertical aspect ratio for cropping (default: 9)
+ * @param onResult Callback function that receives the [ImageCropperResult] after cropping
+ * @return A function that accepts a nullable [Uri] to either crop an existing image or launch the
+ *   gallery picker
+ */
+@Composable
+fun rememberImageCropper(
+    aspectRatioX: Int = 16,
+    aspectRatioY: Int = 9,
+    onResult: (ImageCropperResult) -> Unit
+): (Uri?) -> Unit {
+  val context = LocalContext.current
+
+  // Initialize the image cropper launcher
+  val cropImage =
+      rememberLauncherForActivityResult(CropImageContract()) { result ->
```

Comment:
Here it seems like I have some errors on your branch (It is up to date). I attached a screenshot below. Do you also have these errors on your machine?
<img width="658" alt="Screenshot 2024-11-14 at 17 23 05" src="https://github.com/user-attachments/assets/4014c555-1080-4f89-82b9-616c4ac1d4b1">

Reply:
On my machine everything works. It is possible that some imports have not been updated, sometimes it happens to me. Have you tried to update the imports? Let me know if the problem persists!

Reply:
Fixed, it was a gradle problem. Thanks!